sympy/matrices/expressions/diagonal.py

```diff
@@ -1,14 +1,19 @@
 from __future__ import print_function, division
 
 from sympy.matrices.expressions import MatrixExpr
-from sympy.core import S
-
+from sympy.core import S,Eq
+from sympy.functions.special.tensor_functions import KroneckerDelta
 class DiagonalMatrix(MatrixExpr):
     arg = property(lambda self: self.args[0])
     shape = property(lambda self: (self.arg.shape[0], self.arg.shape[0]))
 
     def _entry(self, i, j):
-        return S.Zero if i != j else self.arg[i, 0]
+        eq=Eq(i,j)
```

Comment:
`i,j` -> `i, j` here and below

Also, please add a test to the test_diagonal.py file:

```
x = MatrixSymbol('x', 3, 3)
ij = DiagonalMatrix(x)[i, j]
assert ij != 0
assert ij.subs({i:0, j:0}) == x[0, 0]
assert ij.subs({i:0, j:1}) == 0
assert ij.subs({i:1, j:1}) == x[1, 1]
```